First round of movr updates

ericharmeling opened the pull request:
This is the very first round of `movr` doc updates. These changes are all for 19.2 doc.

This PR includes the following:
- A section dedicated to explaining the MovR example (movr.md) #5061
- The `movr` command added to the `cockroach demo` and `cockroach workload` command documentation for 19.2 (cockroach-demo.md, cockroach-workload.md) #5062
- Updates to "Learn CockroachDB SQL", to use the `movr` database instead of other generic examples (learn-cockroachdb-sql-movr.md) #5040
- Updates to a few SQL statement example sections, including INSERT (insert.md), SHOW TABLES (show-tables.md), SHOW COLUMNS (show-columns.md), DROP TABLE (drop-table.md), and CREATE TABLE (create-table.md) #5041 ... Many more of these to come!

Note that I've also changed/added several other include files where necessary. I suspect there are plenty of changes that need to be made to this PR upon review. No rush!

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/5075)
<!-- Reviewable:end -->

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/cockroach-workload.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/create-table.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/drop-table.html)
- [v19.2/learn-cockroachdb-sql-movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/learn-cockroachdb-sql-movr.html)
- [v19.2/learn-cockroachdb-sql-overview.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/learn-cockroachdb-sql-overview.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/movr.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/show-columns.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/80c7b619ff1e4aa881dd7a0561a193578bb368c1/v19.2/show-tables.html)

rmloveland commented:
Can you please squash and force push at your leisure?  Will make review easier for me.
(Hoping you don't see this or act on it until after your vacation :-} )

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/cockroach-workload.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/create-table.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/drop-table.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/insert.html)
- [v19.2/learn-cockroachdb-sql-movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/learn-cockroachdb-sql-movr.html)
- [v19.2/learn-cockroachdb-sql-overview.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/learn-cockroachdb-sql-overview.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/movr.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/show-columns.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f273e4626703294d170013a078d8718d46435349/v19.2/show-tables.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/cockroach-workload.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/create-table.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/drop-table.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/insert.html)
- [v19.2/learn-cockroachdb-sql-movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/learn-cockroachdb-sql-movr.html)
- [v19.2/learn-cockroachdb-sql-overview.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/learn-cockroachdb-sql-overview.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/movr.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/show-columns.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/128f805d454ae8f82bf721c2957eda37317a2315/v19.2/show-tables.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/cockroach-workload.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/create-table.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/drop-table.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/insert.html)
- [v19.2/learn-cockroachdb-sql-movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/learn-cockroachdb-sql-movr.html)
- [v19.2/learn-cockroachdb-sql-overview.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/learn-cockroachdb-sql-overview.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/movr.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/show-columns.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9264dc38c97b62e86629ed23fcb94e6396c9c8da/v19.2/show-tables.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/cockroach-workload.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/create-table.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/drop-table.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/insert.html)
- [v19.2/learn-cockroachdb-sql-movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/learn-cockroachdb-sql-movr.html)
- [v19.2/learn-cockroachdb-sql-overview.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/learn-cockroachdb-sql-overview.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/movr.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/show-columns.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c7a13bb962238743052c2a4e1a16b5dbb555a61/v19.2/show-tables.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/cockroach-workload.html)
- [v19.2/create-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/create-index.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/create-table.html)
- [v19.2/delete.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/delete.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/drop-table.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/insert.html)
- [v19.2/learn-cockroachdb-sql-movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/learn-cockroachdb-sql-movr.html)
- [v19.2/learn-cockroachdb-sql-overview.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/learn-cockroachdb-sql-overview.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/movr.html)
- [v19.2/select-clause.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/select-clause.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/show-columns.html)
- [v19.2/show-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/show-index.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/show-tables.html)
- [v19.2/update.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/31054774436e39d739b251673848eefc51df601f/v19.2/update.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/cockroach-workload.html)
- [v19.2/create-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/create-index.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/create-table.html)
- [v19.2/delete.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/delete.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/drop-table.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/insert.html)
- [v19.2/learn-cockroachdb-sql-movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/learn-cockroachdb-sql-movr.html)
- [v19.2/learn-cockroachdb-sql-overview.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/learn-cockroachdb-sql-overview.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/movr.html)
- [v19.2/select-clause.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/select-clause.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/show-columns.html)
- [v19.2/show-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/show-index.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/show-tables.html)
- [v19.2/update.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/17a452b923de7564ed505c342804c9d7ddea6e47/v19.2/update.html)

rmloveland commented on _includes/sidebar-data-v19.2.json:
```diff
@@ -43,7 +43,7 @@
             "title": "Learn CockroachDB SQL",
             "urls": [
-              "/${VERSION}/learn-cockroachdb-sql.html"
+              "/${VERSION}/learn-cockroachdb-sql-movr.html"
```
We should not do this IMO for several reasons: (1) we throw away all our PageRank on the existing page built up over several years, (2) the existing page becomes an orphan since AFAICT it's still in the site.  We should instead update the existing "Learn CRDB SQL" page with the MovR content IMO

ericharmeling replied:
Both totally legitimate reasons to just update the file with the same name rather than create an additional file and have two separate files! I'll fix this.

rmloveland commented on _includes/v19.2/sql/movr-start.md:
```diff
@@ -0,0 +1,6 @@
+- Run [`cockroach demo movr`](cockroach-demo.html) to open an interactive SQL shell to a temporary, in-memory cluster with the `movr` database preloaded and set as the [current database](sql-name-resolution.html#current-database).
+- Use [`cockroach workload`](cockroach-workload.html):
+ 1. Start up a [secure](secure-a-cluster.html) or [insecure](start-a-local-cluster.html) local cluster.
+ 1. Run `cockroach workload init movr` with the appropriate flags and [connection string](connection-parameters.html) to initialize and populate the `movr` database on your running cluster.
```
Could we show the exact command string here for at least the simplest case?  And keep the links for people who need something more complicated

ericharmeling replied:
Sure thing!

rmloveland commented on _includes/v19.2/sql/movr-start.md:
```diff
@@ -0,0 +1,6 @@
+- Use [`cockroach workload`](cockroach-workload.html):
+ 1. Start up a [secure](secure-a-cluster.html) or [insecure](start-a-local-cluster.html) local cluster.
+ 1. Run `cockroach workload init movr` with the appropriate flags and [connection string](connection-parameters.html) to initialize and populate the `movr` database on your running cluster.
+ 1. Open an interactive SQL shell to the cluster with the [`cockroach sql`](use-the-built-in-sql-client.html) command.
```
Same comment as in some other places re: could we show the exact `$ cockroach sql --insecure ...` command?  Would make it easier for users to copypasta and get through the instructions

ericharmeling replied:
Sure thing! Question: Should we be providing copy-paste commands for users trying to start a secure cluster, or just for insecure clusters? We would have to make the assumption that folks have already created certs, etc, or we would have to provide sample commands for all of those steps...

rmloveland replied:
As we discussed, my vote is for simple insecure-only, since if you're using secure we can't really know what your connection string is.

rmloveland commented on _includes/v19.2/sql/movr-start.md:
```diff
@@ -0,0 +1,6 @@
+ 1. Start up a [secure](secure-a-cluster.html) or [insecure](start-a-local-cluster.html) local cluster.
+ 1. Run `cockroach workload init movr` with the appropriate flags and [connection string](connection-parameters.html) to initialize and populate the `movr` database on your running cluster.
+ 1. Open an interactive SQL shell to the cluster with the [`cockroach sql`](use-the-built-in-sql-client.html) command.
+ 1. Set `movr` as the [current database](sql-name-resolution.html#current-database) for the session with the [`SET`](set-vars.html) statement.
```
Same comment as before re: could we show the explicit command for copypasta?  also, I think it's easiest to `> USE movr;` instead of `SET ...`

ericharmeling replied:
Done!

rmloveland commented on _includes/v19.2/sql/movr-statements-no-data.md:
```diff
@@ -0,0 +1,10 @@
+The following examples use MovR, a fictional vehicle-sharing application, to demonstrate CockroachDB SQL statements. For more information about the MovR example application, see [MovR: A Global Ride-sharing App](movr.html).
```
I think we should standardize on either ride-sharing or vehicle-sharing.  AFAICT it's really more "vehicle sharing" since there is no drivers table, etc.  Is that correct?

ericharmeling replied:
I 100% agree on this. It is, in fact, a vehicle-sharing app. I confirmed this with Nate last week. I plan to make this global change a part of the PR.

rmloveland commented on _includes/v19.2/sql/movr-statements.md:
```diff
@@ -0,0 +1,5 @@
+
+To follow along with the example statements, do one of the following:
+
+{% include {{page.version.version}}/sql/movr-start.md %}
```
Is it necessary to have the double-layer of includes here?  Maybe I'm not understanding the whole structure of this.
General comment on the include strategy: I wonder also if we could get away with having one `movr-statements.md` include rather than having one each for with/without data.  Maybe there's a way to write it up as a choice, like "in case A, do this; in case B, do this"?

ericharmeling replied:
The intro sentence really only applies to SQL statement examples (e.g. insert.md), but the movr-start.md instructions apply to all examples where we use the MovR dataset (e.g. learn-cockroachdb-sql.md). If we could use conditional includes, that would be amazing. Candidly, I don't know how to do that... I'll follow up offline about this.

rmloveland replied:
SGTM

rmloveland commented on v19.2/create-table.md:
```diff
@@ -103,140 +105,132 @@ In CockroachDB, every table requires a [primary key](primary-key.html). If one i
 ### Create a table (primary key defined)
 
-In this example, we create a table with three columns. One column is the [`PRIMARY KEY`](primary-key.html), another is given the [`UNIQUE` constraint](unique.html), and the third has no constraints. The `PRIMARY KEY` and column with the `UNIQUE` constraint are automatically indexed.
+In this example, we create the `users` table, but with some column [constraints](constraints.html). One column is the [`PRIMARY KEY`](primary-key.html), another is given the [`UNIQUE` constraint](unique.html), and another has both a `UNIQUE` constraint and a [`CHECK`](check.html) constraint that limits the length of the string. The `PRIMARY KEY` and column with the `UNIQUE` constraint are automat ...
```
I think our convention is to refer to these things in prose without upper-case or monospace as a "primary key", "unique constraint", "check constraint", etc., except in cases where we are directly referring to SQL statement text.  Although I was not able to find this in [our style guide](https://github.com/cockroachdb/docs/blob/master/STYLE.md), so maybe we need an update there.  But I'm pretty sure this is our convention based on convos I've had during PR reviews.

ericharmeling replied:
Makes sense to me!

ericharmeling replied:
Do we typically keep the reference links to other doc pages though?

rmloveland replied:
Yes - I am only referring to changing the text.

rmloveland commented on v19.2/create-table.md:
```diff
@@ -103,140 +105,132 @@ In CockroachDB, every table requires a [primary key](primary-key.html). If one i
+users      | users_name_key |   false    |            1 | name        | ASC       |  false  |  false
+users      | users_name_key |   false    |            2 | id          | ASC       |  false  |   true
+users      | users_dl_key   |   false    |            1 | dl          | ASC       |  false  |  false
+users      | users_dl_key   |   false    |            2 | id          | ASC       |  false  |   true
```
This table appears to be a little out of alignment. #nitpicking

ericharmeling replied:
Great catch!

rmloveland commented on v19.2/create-table.md:
```diff
@@ -264,74 +258,108 @@ In this example, we use `ON DELETE CASCADE` (i.e., when row referenced by a fore
-|            |                                                                          |
-|            | )                                                                        |
-+------------+--------------------------------------------------------------------------+
+> SHOW CREATE vehicles;
```
This could be slightly more specific as `SHOW CREATE TABLE` but IDK if it matters.  Either way it works!

ericharmeling replied:
Done.

rmloveland commented on v19.2/create-table.md:
```diff
@@ -264,74 +258,108 @@ In this example, we use `ON DELETE CASCADE` (i.e., when row referenced by a fore
+~~~
+                   id                  |  city   | type |               owner_id               | creation_time | status | current_location | ext
++--------------------------------------+---------+------+--------------------------------------+---------------+--------+------------------+------+
+  fc6f7a8c-4ba9-42e1-9c37-7be3c906050c | seattle | NULL | 26da1fce-59e1-4290-a786-9068242dd195 | NULL          | NULL   | NULL             | NULL
```
Interesting, so the DB schema doesn't auto-fill a creation time.  Huh.  Not material to this review, just a possible MovR code/DB update for the future, perhaps?  TBH it strikes me as odd that this field, status, and current location are allowed to be NULL.  Particularly location and status.

ericharmeling replied:
I agree! There are a lot of gaps in the data generator like this... I am compiling a list of improvements to the dataset/generator. Setting the default values as non-null, generated values is definitely among on that list.

rmloveland commented on v19.2/drop-table.md:
```diff
@@ -62,52 +65,82 @@ DROP TABLE
 ### Remove a table and dependent objects with `CASCADE`
 
-In this example, a view depends on the table being dropped. Therefore, it's only possible to drop the table while simultaneously dropping the dependent view using `CASCADE`.
+In this example, a foreign key from a different table references the table being dropped. Therefore, it's only possible to drop the table while simultaneously dropping the dependent foreign key constraint using `CASCADE`.
```
It might be nice to make the first mention of "foreign key" into a link to the Foreign Key Constraint doc.

ericharmeling replied:
Done.

rmloveland commented on v19.2/drop-table.md:
```diff
@@ -62,52 +65,82 @@ DROP TABLE
+pq: "users" is referenced by foreign key from table "vehicles"
 ~~~
 
+You can use the [`SHOW CREATE`](show-create.html) statement to see how the columns in a table are created, including data types, default values, indexes, and constraints.
```
It might be nice to preface this with some context, something like "To see how 'users' is referenced from 'vehicles', use the `SHOW CREATE` ..."

ericharmeling replied:
Done.

rmloveland commented on v19.2/drop-table.md:
```diff
@@ -116,15 +149,41 @@ DROP TABLE
+  vehicles
+(4 rows)
+~~~
+
```
It might be good to add some explication here between these statements like "Now, we can verify the FK has been removed by running SHOW CREATE again and seeing that the XXX_fk is gone" ... or similar.

ericharmeling replied:
Done!

rmloveland commented on v19.2/insert.md:
```diff
@@ -67,68 +67,81 @@ key. Using `ON CONFLICT` is therefore more flexible.
-    balance DECIMAL
-);
-~~~
+{% include {{page.version.version}}/sql/movr-statements.md %}
```
Left comments on `movr-statements.md` already, but wanna reiterate question of whether we can revamp to add "logic" to the text so we don't need `movr-statements-no-data.md`.

ericharmeling replied:
Sure thing - let's talk about this.

rmloveland commented on v19.2/insert.md:
```diff
@@ -67,68 +67,81 @@ key. Using `ON CONFLICT` is therefore more flexible.
+  0a3d70a3-d70a-4d80-8000-000000000002 | new york | Judy White       | 18580 Rosario Ville Apt. 61 | 2597958636
+  0f5c28f5-c28f-4c00-8000-000000000003 | new york | Devin Jordan     | 81127 Angela Ferry Apt. 8   | 5614075234
+  147ae147-ae14-4b00-8000-000000000004 | new york | Catherine Nelson | 1149 Lee Alley              | 0792553487
+  c28f5c28-f5c2-4000-8000-000000000026 | new york | Petee            | 101 5th Ave                 | 1234567890
```
Movr comment, not docs comment: we should update the credit card handling to be more realistic at some point, hashed, elliptic curved, or "just use Stripe", or whatever the correct behavior should be - not plaintext.

ericharmeling replied:
Agreed. I'll add this to the notes.

rmloveland commented on v19.2/insert.md:
```diff
@@ -439,204 +471,195 @@ IDs:
-{ id: '190019066706952193' }
-{ id: '190019066706984961' }
+{ id: 'cdd379e3-2d0b-4622-8ba8-4f0a1edfbc8e' }
+{ id: '4224b360-b1b0-4e4d-aba2-a35c64cdf404' }
```
All of these code sample updates LGTM at first blush.  After I get through the initial prose/doc review, I'll take a second pass through where I check all the SQL statements and code snippets.  Gosh we need automated testing :-)

ericharmeling replied:
Sounds good! +1

rmloveland commented on v19.2/learn-cockroachdb-sql-movr.md:
```diff
@@ -0,0 +1,381 @@
+---
+title: Learn CockroachDB SQL
```
I don't understand why this is a new file named `learn-cockroachdb-sql-movr.md` and not an update to the existing `learn-cockroachdb-sql.md`?  Aha, it looks like you pinged me in #5040 but I didn't get an email, weird.  My opinion is that we should do the minimal update to edit the existing "Learn CockroachDB SQL" page to make it "movr-compatible", since I think making "Learn CockroachDB SQL" a family of pages with different options for starting the database is adding complexity to the "user journey", and using "cockroach demo" vs. spinning up a local cluster is sort of a separate issue we should decide how to address / standardize in our docs IMO.  But let's discuss!  We should def get Jesse's opinion when he returns from vacation, I'm curious about his thoughts also.

ericharmeling replied:
I think you are right on this... We should be aiming to simplify docs, especially "Learn CockroachDB" docs and others of that ilk. I'll simplify this just as you suggest.

rmloveland commented on v19.2/learn-cockroachdb-sql-movr.md:
```diff
@@ -0,0 +1,381 @@
+build_for: [standard, managed]
+---
+
+This tutorial walks you through some of the most essential CockroachDB SQL statements, using the `movr` dataset. For a complete list of supported SQL statements and related details, see [SQL Statements](sql-statements.html). For more information about the `movr` dataset and the MovR example application, see [MovR: A Global Ride-sharing App](movr.html).
```
These 2 finishing sentences would be easier to scan if they were each on their own line at the end of this intro

ericharmeling replied:
Done.

rmloveland commented on v19.2/learn-cockroachdb-sql-movr.md:
```diff
@@ -0,0 +1,381 @@
+This tutorial walks you through some of the most essential CockroachDB SQL statements, using the `movr` dataset. For a complete list of supported SQL statements and related details, see [SQL Statements](sql-statements.html). For more information about the `movr` dataset and the MovR example application, see [MovR: A Global Ride-sharing App](movr.html).
+
+{{site.data.alerts.callout_info}}
+CockroachDB aims to provide standard SQL with extensions, but some standard SQL functionality is not yet available. See our [SQL Feature Support](sql-feature-support.html) page for more details.
```
I think this note was already here, but I've filed https://github.com/cockroachdb/docs/issues/5125 to try to remove it

ericharmeling replied:
OK. I'll go ahead and remove it from here.

rmloveland commented on v19.2/learn-cockroachdb-sql.md:
```diff
@@ -1,6 +1,6 @@
 ---
-title: Learn CockroachDB SQL
-summary: Learn some of the most essential CockroachDB SQL statements.
+title: Learn CockroachDB SQL with a Running Cluster
```
Hm, from looking at the auto-built test site, it looks like this file is an orphan from the TOC now.  I left a comment above on the TOC JSON that is basically: I don't think we should add a new "Learn CRDB SQL MovR" page, I think we should edit the existing page with the MovR updates.  Reasoning is explained in that comment.

ericharmeling replied:
Done.

rmloveland commented on v19.2/movr.md:
```diff
@@ -0,0 +1,85 @@
+
+- The `movr` database, which contains several tables that hold vehicle, user, and ride data. The `movr` database is built into `cockroach demo` and `cockroach workload`.
+- The `movr` dataset, which consists of rows of data that populate tables in the `movr` database. The `movr` dataset is built into `cockroach demo` and `cockroach workload`.
+- The MovR application is a fully-functional vehicle-sharing application, written in Python. All of MovR application source code is open-source, and available on the [movr](https://github.com/cockroachdb/movr) GitHub repository.
```
All of the mentions (or at least the first mention) of `cockroach demo` and `cockroach workload` could link to the docs for those commands.

ericharmeling replied:
Done!

rmloveland commented on v19.2/movr.md:
```diff
@@ -0,0 +1,85 @@
+
+## The `movr` Database
+
+Three tables in the `movr` database store most of the user, vehicle, and ride data for MovR:
```
It may be correct that they store "most of" the data, but perhaps for simplicity of introducing this it could be glossed over?  Genuinely asking because I don't actually know movr that well.  Alternatively, if we do keep it in, IMO we should say where "the rest of it" is.

ericharmeling replied:
Good call. I removed that "most of" part. There is a sentence further down that lists the other tables (currently) in the database.

rmloveland commented on v19.2/movr.md:
```diff
@@ -0,0 +1,85 @@
+
+## Using `cockroach` Commands to Generate Schemas and Data for MovR
+
+Use `cockroach` commands to load the `movr` database and dataset that are built into the CockroachDB binary.
```
More of a style thing, but I'd probably edit this slightly to say "Use the following cockroach commands ..." and then make the following 2 items a bulleted list.

ericharmeling replied:
Done!

rmloveland commented on v19.2/movr.md:
```diff
@@ -0,0 +1,85 @@
+
+## Using the MovR Application
+
+The workflow for MovR is as follows:
```
I was a little confused by this sentence until I read the steps that follow.  It might be a little clearer to say something like "at a high level, the movr app works as follows" (or something much better than that obv :-} )

ericharmeling replied:
Done.

rmloveland commented on v19.2/movr.md:
```diff
@@ -0,0 +1,85 @@
+
+    ~~~ sql
+    > UPDATE rides SET end_address = <value> ...
+    ~~~
```
This section is great!  The mapping of use cases to SQL is so easy to follow / understand.

ericharmeling replied:
This was shamelessly copied from https://www.cockroachlabs.com/docs/stable/demo-geo-partitioning.html ! :)

rmloveland commented on v19.2/movr.md:
```diff
@@ -0,0 +1,85 @@
+
+## Extended Examples
+
+MovR scales as a geo-partitioned application and database that you can deploy using Docker images across a multi-region cluster deployment. For a full example, see [Geo-Partitioning for Fast Reads and Writes in a Multi-Region Cluster](demo-geo-partitioning.html).
```
This could possibly be a bit clearer if the latter 2 sentences were each on their own line for easier scanning, as well as revised so their clause structures approximately match a "for more info about X, see Y" formulation, e.g.,
"For an example showing how to tune geo-partitioning for fast reads and writes ..., see ..."
"For a tutorial showing how to tune CockroachDB's performance, see ..."
Writer nerd thing: This type of phrasing is taken from the MSFT style guide 3rd ed. which says
"Information about why the link is interesting should precede the link itself."
Bonus: the reader knows quickly whether they need to read to the end of the sentence or not.
(PS I love that book, it's kinda weird and dated but very cool IMO)

ericharmeling replied:
Done. This is also just awesome general writing advice! Thanks!

I'll have to check out this book...

rmloveland commented on v19.2/movr.md:
```diff
@@ -0,0 +1,85 @@
+
+MovR scales as a geo-partitioned application and database that you can deploy using Docker images across a multi-region cluster deployment. For a full example, see [Geo-Partitioning for Fast Reads and Writes in a Multi-Region Cluster](demo-geo-partitioning.html).
+
+You can also learn more about performance tuning in CockroachDB with the MovR example: [Performance Tuning](performance-tuning.html).
```
I think we should add a 'Related Topics' section to the bottom of this page similar to the one found on most other pages.  It could link to various other movr-related things

ericharmeling replied:
Done!

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/cockroach-workload.html)
- [v19.2/create-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/create-index.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/create-table.html)
- [v19.2/delete.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/delete.html)
- [v19.2/demo-geo-partitioning.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/demo-geo-partitioning.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/drop-table.html)
- [v19.2/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/experimental-features.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/insert.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/movr.html)
- [v19.2/select-clause.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/select-clause.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/show-columns.html)
- [v19.2/show-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/show-index.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/show-tables.html)
- [v19.2/update.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/update.html)
- [v19.2/window-functions.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/v19.2/window-functions.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/cockroach-workload.html)
- [v19.2/create-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/create-index.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/create-table.html)
- [v19.2/delete.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/delete.html)
- [v19.2/demo-geo-partitioning.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/demo-geo-partitioning.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/drop-table.html)
- [v19.2/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/experimental-features.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/insert.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/movr.html)
- [v19.2/select-clause.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/select-clause.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/show-columns.html)
- [v19.2/show-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/show-index.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/show-tables.html)
- [v19.2/update.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/update.html)
- [v19.2/window-functions.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b538364ed9ed32f4ea9eb4a2b4b39d3cadbab282/v19.2/window-functions.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/cockroach-workload.html)
- [v19.2/create-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/create-index.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/create-table.html)
- [v19.2/delete.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/delete.html)
- [v19.2/demo-geo-partitioning.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/demo-geo-partitioning.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/drop-table.html)
- [v19.2/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/experimental-features.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/insert.html)
- [v19.2/learn-cockroachdb-sql.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/learn-cockroachdb-sql.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/movr.html)
- [v19.2/select-clause.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/select-clause.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/show-columns.html)
- [v19.2/show-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/show-index.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/show-tables.html)
- [v19.2/update.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/update.html)
- [v19.2/window-functions.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d41b9169a6d25e48f7aa3e57230a44745030b287/v19.2/window-functions.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/cockroach-workload.html)
- [v19.2/create-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/create-index.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/create-table.html)
- [v19.2/delete.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/delete.html)
- [v19.2/demo-geo-partitioning.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/demo-geo-partitioning.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/drop-table.html)
- [v19.2/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/experimental-features.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/insert.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/movr.html)
- [v19.2/select-clause.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/select-clause.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/show-columns.html)
- [v19.2/show-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/show-index.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/show-tables.html)
- [v19.2/update.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/update.html)
- [v19.2/window-functions.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/window-functions.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/cockroach-workload.html)
- [v19.2/create-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/create-index.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/create-table.html)
- [v19.2/delete.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/delete.html)
- [v19.2/demo-geo-partitioning.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/demo-geo-partitioning.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/drop-table.html)
- [v19.2/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/experimental-features.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/insert.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/movr.html)
- [v19.2/select-clause.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/select-clause.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/show-columns.html)
- [v19.2/show-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/show-index.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/show-tables.html)
- [v19.2/update.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/update.html)
- [v19.2/window-functions.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/window-functions.html)

jseldess commented:
___
*[v19.2/movr.md, line 13 at r3](https://reviewable.io/reviews/cockroachdb/docs/5075#-LmF8H0WDGeuR63JA-Yn:-LmF8H0WDGeuR63JA-Yo:bbx1bt5) ([raw file](https://github.com/cockroachdb/docs/blob/b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/movr.md#L13)):*
> ```Markdown
> The MovR example consists of three components.
> 
> - The `movr` database, which contains several tables that hold vehicle, user, and ride data. The `movr` database is built into [`cockroach demo`](cockroach-demo.html) and [`cockroach workload`](cockroach-workload.html).
> ```

On another review, I think this first bullet isn't necessary. Focusing on this as a dataset, which implies a database, and an application, seems clearer.


<!-- Sent from Reviewable.io -->

jseldess commented:
___
*[_includes/sidebar-data-v19.2.json, line 126 at r2](https://reviewable.io/reviews/cockroachdb/docs/5075#-Lll_8rS7_MZhAzfYqsh:-LmF8UcB9s90yCQCwo32:bqvct4i) ([raw file](https://github.com/cockroachdb/docs/blob/0cf4fe269ac9d714ad93eb0d5e01eeed96809384/_includes/sidebar-data-v19.2.json#L126)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Valid point, but despite it also being a static dataset, I think "Sample Applications" focuses on the fact that it is based on a fictional application. I like it how you have it, and I like imagining that we might add more sample applications as well. Maybe show it in the sidenav as `MovR: vehicle-sharing app" or something?
</blockquote></details>

Thinking about this more. We could also name this section just "Workloads" and consider featuring more of all of the workloads and datasets built into `cockroach workload` and `cockroach demo` (separately, of course). I'm thinking YCSB and TPCC would be particularly useful to call out as workloads. Thoughts?


<!-- Sent from Reviewable.io -->

jseldess commented:
@ericharmeling, teamcity is failing due to these broken links:

```
[21:39:34][Step 1/1] ========================================================================
[21:39:34][Step 1/1] running in concurrent mode, this is experimental
[21:39:35][Step 1/1] b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/indexes.html
[21:39:35][Step 1/1]   hash does not exist --- b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/indexes.html --> create-table.html#create-a-table-primary-key-defined
[21:39:35][Step 1/1] b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/add-constraint.html
[21:39:35][Step 1/1]   hash does not exist --- b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/add-constraint.html --> create-table.html#create-a-table-primary-key-defined
[21:39:36][Step 1/1] ========================================================================
```

Let me know if you need any help resolving them.

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/

Edited pages:
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/cockroach-workload.html)
- [v19.2/create-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/create-index.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/create-table.html)
- [v19.2/delete.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/delete.html)
- [v19.2/demo-geo-partitioning.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/demo-geo-partitioning.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/drop-table.html)
- [v19.2/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/experimental-features.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/insert.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/movr.html)
- [v19.2/select-clause.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/select-clause.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/show-columns.html)
- [v19.2/show-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/show-index.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/show-tables.html)
- [v19.2/update.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/update.html)
- [v19.2/window-functions.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae841b9fe3b8e37f6b982fb79d391a46bb41a2b8/v19.2/window-functions.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/

Edited pages:
- [v19.2/add-constraint.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/add-constraint.html)
- [v19.2/cockroach-demo.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/cockroach-demo.html)
- [v19.2/cockroach-workload.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/cockroach-workload.html)
- [v19.2/create-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/create-index.html)
- [v19.2/create-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/create-table.html)
- [v19.2/delete.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/delete.html)
- [v19.2/demo-geo-partitioning.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/demo-geo-partitioning.html)
- [v19.2/drop-table.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/drop-table.html)
- [v19.2/experimental-features.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/experimental-features.html)
- [v19.2/indexes.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/indexes.html)
- [v19.2/insert.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/insert.html)
- [v19.2/movr.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/movr.html)
- [v19.2/select-clause.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/select-clause.html)
- [v19.2/show-columns.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/show-columns.html)
- [v19.2/show-index.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/show-index.html)
- [v19.2/show-tables.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/show-tables.html)
- [v19.2/update.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/update.html)
- [v19.2/window-functions.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d5a9ad32cafb4dc5177a5815edf6f2d3285eff17/v19.2/window-functions.html)

ericharmeling commented:
> @ericharmeling, teamcity is failing due to these broken links:
> 
> ```
> [21:39:34][Step 1/1] ========================================================================
> [21:39:34][Step 1/1] running in concurrent mode, this is experimental
> [21:39:35][Step 1/1] b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/indexes.html
> [21:39:35][Step 1/1]   hash does not exist --- b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/indexes.html --> create-table.html#create-a-table-primary-key-defined
> [21:39:35][Step 1/1] b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/add-constraint.html
> [21:39:35][Step 1/1]   hash does not exist --- b1544f5348dc974a88daee7be262c0dcbb0db432/v19.2/add-constraint.html --> create-table.html#create-a-table-primary-key-defined
> [21:39:36][Step 1/1] ========================================================================
> ```
> 
> Let me know if you need any help resolving them.

Fixed the broken links... These happened because I removed the `CREATE TABLE` example that doesn't define a primary key in response to #4912.
